service/internal/cache/cache_test.go

```diff
@@ -0,0 +1,54 @@
+package cache
+
+import (
+	"testing"
+	"time"
+)
+
+func Test_floatCache_Set(t *testing.T) {
+	c := NewFloat64Cache(time.Minute * 1)
+
+	testCases := []struct {
+		key           string
+		value         float64
+		expectedValue float64
+	}{
+		{
+			key:           "key",
+			value:         0,
+			expectedValue: 0,
+		},
+		{
+			key:           "key",
+			value:         3.1416,
+			expectedValue: 3.1416,
+		},
+	}
+
+	for _, tc := range testCases {
+		c.Set(tc.key, tc.value)
+
+		value, ok := c.Get(tc.key)
+		if !ok {
+			t.Fatalf("cache key must exist")
+		}
+		if value != tc.expectedValue {
+			t.Fatalf("cache value == %v, want %v", value, tc.expectedValue)
+		}
+	}
+}
+
+func Test_floatCache_Expire(t *testing.T) {
+	c := NewFloat64Cache(time.Second * 1)
+
+	key := "pi"
+	pi := 3.14
+	c.Set(key, pi)
+
+	time.Sleep(time.Second * 1)
```

Comment:
This test delays 1 sec the entire test suite, do you think it is worth?

Reply:
I think time based tests are always not so cool. When you think this test ensures important functionality then let's go with it but I am not big on it.